# <center> INX FUTURE INC EMPLOYEE PERFORMANCE ANALYSIS

## <center> BUISNESS CASE:BASED ON GIVEN FEATURE OF DATASET WE NEED TO PREDICT THE PERFOMANCE RATING OF EMPLOYEE

***Importing the necessary libraries***

In [1]:
import pandas as pd
import numpy as np

In [2]:
# importing the csv file
data = pd.read_excel('../../data/INX_Future_Inc_Employee_Performance_CDS_Project2_Data_V1.8 (1).xls')

In [3]:
# setting display of columns to max
pd.set_option("display.max_columns", None)

## Basic Checks

In [4]:
# showing fisrt five row of the dataset
data.head()

EmpNumber  Age Gender EducationBackground MaritalStatus    EmpDepartment  \
0  E1001000   32   Male           Marketing        Single            Sales   
1  E1001006   47   Male           Marketing        Single            Sales   
2  E1001007   40   Male       Life Sciences       Married            Sales   
3  E1001009   41   Male     Human Resources      Divorced  Human Resources   
4  E1001010   60   Male           Marketing        Single            Sales   

        EmpJobRole BusinessTravelFrequency  DistanceFromHome  \
0  Sales Executive           Travel_Rarely                10   
1  Sales Executive           Travel_Rarely                14   
2  Sales Executive       Travel_Frequently                 5   
3          Manager           Travel_Rarely                10   
4  Sales Executive           Travel_Rarely                16   

   EmpEducationLevel  EmpEnvironmentSatisfaction  EmpHourlyRate  \
0                  3                           4             55   
1                  4                           4             42   
2                  4                           4             48   
3                  4                           2             73   
4                  4                           1             84   

   EmpJobInvolvement  EmpJobLevel  EmpJobSatisfaction  NumCompaniesWorked  \
0                  3            2                   4                   1   
1                  3            2                   1                   2   
2                  2            3                   1                   5   
3                  2            5                   4                   3   
4                  3            2                   1                   8   

  OverTime  EmpLastSalaryHikePercent  EmpRelationshipSatisfaction  \
0       No                        12                            4   
1       No                        12                            4   
2      Yes                        21                            3   
3       No                        15                            2   
4       No                        14                            4   

   TotalWorkExperienceInYears  TrainingTimesLastYear  EmpWorkLifeBalance  \
0                          10                      2                   2   
1                          20                      2                   3   
2                          20                      2                   3   
3                          23                      2                   2   
4                          10                      1                   3   

   ExperienceYearsAtThisCompany  ExperienceYearsInCurrentRole  \
0                            10                             7   
1                             7                             7   
2                            18                            13   
3                            21                             6   
4                             2                             2   

   YearsSinceLastPromotion  YearsWithCurrManager Attrition  PerformanceRating  
0                        0                     8        No                  3  
1                        1                     7        No                  3  
2                        1                    12        No                  4  
3                       12                     6        No                  3  
4                        2                     2        No                  3

In [5]:
# set the last five row of the dataset
data.tail()

EmpNumber  Age  Gender EducationBackground MaritalStatus EmpDepartment  \
1195   E100992   27  Female             Medical      Divorced         Sales   
1196   E100993   37    Male       Life Sciences        Single   Development   
1197   E100994   50    Male             Medical       Married   Development   
1198   E100995   34  Female             Medical        Single  Data Science   
1199   E100998   24  Female       Life Sciences        Single         Sales   

            EmpJobRole BusinessTravelFrequency  DistanceFromHome  \
1195   Sales Executive       Travel_Frequently                 3   
1196  Senior Developer           Travel_Rarely                10   
1197  Senior Developer           Travel_Rarely                28   
1198    Data Scientist           Travel_Rarely                 9   
1199   Sales Executive           Travel_Rarely                 3   

      EmpEducationLevel  EmpEnvironmentSatisfaction  EmpHourlyRate  \
1195                  1                           4             71   
1196                  2                           4             80   
1197                  1                           4             74   
1198                  3                           4             46   
1199                  2                           1             65   

      EmpJobInvolvement  EmpJobLevel  EmpJobSatisfaction  NumCompaniesWorked  \
1195                  4            2                   4                   1   
1196                  4            1                   4                   3   
1197                  4            1                   3                   1   
1198                  2            3                   2                   1   
1199                  3            2                   3                   9   

     OverTime  EmpLastSalaryHikePercent  EmpRelationshipSatisfaction  \
1195      Yes                        20                            2   
1196       No                        17                            1   
1197      Yes                        11                            3   
1198       No                        14                            2   
1199       No                        14                            1   

      TotalWorkExperienceInYears  TrainingTimesLastYear  EmpWorkLifeBalance  \
1195                           6                      3                   3   
1196                           4                      2                   3   
1197                          20                      3                   3   
1198                           9                      3                   4   
1199                           4                      3                   3   

      ExperienceYearsAtThisCompany  ExperienceYearsInCurrentRole  \
1195                             6                             5   
1196                             1                             0   
1197                            20                             8   
1198                             8                             7   
1199                             2                             2   

      YearsSinceLastPromotion  YearsWithCurrManager Attrition  \
1195                        0                     4        No   
1196                        0                     0        No   
1197                        3                     8        No   
1198                        7                     7        No   
1199                        2                     0       Yes   

      PerformanceRating  
1195                  4  
1196                  3  
1197                  3  
1198                  3  
1199                  2

In [6]:
# displaying the the shape of the dataset
data.shape

(1200, 28)

In [7]:
# displaying the total rows of the columns
data.shape[0]

1200

In [8]:
# displaying the total columns in dataset
data.shape[1]

28

In [9]:
# total columns present in the dataset
data.columns

Index(['EmpNumber', 'Age', 'Gender', 'EducationBackground', 'MaritalStatus',
       'EmpDepartment', 'EmpJobRole', 'BusinessTravelFrequency',
       'DistanceFromHome', 'EmpEducationLevel', 'EmpEnvironmentSatisfaction',
       'EmpHourlyRate', 'EmpJobInvolvement', 'EmpJobLevel',
       'EmpJobSatisfaction', 'NumCompaniesWorked', 'OverTime',
       'EmpLastSalaryHikePercent', 'EmpRelationshipSatisfaction',
       'TotalWorkExperienceInYears', 'TrainingTimesLastYear',
       'EmpWorkLifeBalance', 'ExperienceYearsAtThisCompany',
       'ExperienceYearsInCurrentRole', 'YearsSinceLastPromotion',
       'YearsWithCurrManager', 'Attrition', 'PerformanceRating'],
      dtype='object')

In [10]:
# dataset information of the dataset
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 28 columns):
 #   Column                        Non-Null Count  Dtype 
---  ------                        --------------  ----- 
 0   EmpNumber                     1200 non-null   object
 1   Age                           1200 non-null   int64 
 2   Gender                        1200 non-null   object
 3   EducationBackground           1200 non-null   object
 4   MaritalStatus                 1200 non-null   object
 5   EmpDepartment                 1200 non-null   object
 6   EmpJobRole                    1200 non-null   object
 7   BusinessTravelFrequency       1200 non-null   object
 8   DistanceFromHome              1200 non-null   int64 
 9   EmpEducationLevel             1200 non-null   int64 
 10  EmpEnvironmentSatisfaction    1200 non-null   int64 
 11  EmpHourlyRate                 1200 non-null   int64 
 12  EmpJobInvolvement             1200 non-null   int64 
 13  EmpJobLevel       

- from the above we can see that there are to 28 columns in the the dataset having no null values present in the dataset
- there are total 9 columns which are categorical in nature and rest of them are numerical in nature

### Showing the statical feature of the dataset

In [11]:
data.describe()

Age  DistanceFromHome  EmpEducationLevel  \
count  1200.000000       1200.000000         1200.00000   
mean     36.918333          9.165833            2.89250   
std       9.087289          8.176636            1.04412   
min      18.000000          1.000000            1.00000   
25%      30.000000          2.000000            2.00000   
50%      36.000000          7.000000            3.00000   
75%      43.000000         14.000000            4.00000   
max      60.000000         29.000000            5.00000   

       EmpEnvironmentSatisfaction  EmpHourlyRate  EmpJobInvolvement  \
count                 1200.000000    1200.000000        1200.000000   
mean                     2.715833      65.981667           2.731667   
std                      1.090599      20.211302           0.707164   
min                      1.000000      30.000000           1.000000   
25%                      2.000000      48.000000           2.000000   
50%                      3.000000      66.000000           3.000000   
75%                      4.000000      83.000000           3.000000   
max                      4.000000     100.000000           4.000000   

       EmpJobLevel  EmpJobSatisfaction  NumCompaniesWorked  \
count  1200.000000         1200.000000         1200.000000   
mean      2.067500            2.732500            2.665000   
std       1.107836            1.100888            2.469384   
min       1.000000            1.000000            0.000000   
25%       1.000000            2.000000            1.000000   
50%       2.000000            3.000000            2.000000   
75%       3.000000            4.000000            4.000000   
max       5.000000            4.000000            9.000000   

       EmpLastSalaryHikePercent  EmpRelationshipSatisfaction  \
count               1200.000000                  1200.000000   
mean                  15.222500                     2.725000   
std                    3.625918                     1.075642   
min                   11.000000                     1.000000   
25%                   12.000000                     2.000000   
50%                   14.000000                     3.000000   
75%                   18.000000                     4.000000   
max                   25.000000                     4.000000   

       TotalWorkExperienceInYears  TrainingTimesLastYear  EmpWorkLifeBalance  \
count                 1200.000000            1200.000000         1200.000000   
mean                    11.330000               2.785833            2.744167   
std                      7.797228               1.263446            0.699374   
min                      0.000000               0.000000            1.000000   
25%                      6.000000               2.000000            2.000000   
50%                     10.000000               3.000000            3.000000   
75%                     15.000000               3.000000            3.000000   
max                     40.000000               6.000000            4.000000   

       ExperienceYearsAtThisCompany  ExperienceYearsInCurrentRole  \
count                   1200.000000                   1200.000000   
mean                       7.077500                      4.291667   
std                        6.236899                      3.613744   
min                        0.000000                      0.000000   
25%                        3.000000                      2.000000   
50%                        5.000000                      3.000000   
75%                       10.000000                      7.000000   
max                       40.000000                     18.000000   

       YearsSinceLastPromotion  YearsWithCurrManager  PerformanceRating  
count              1200.000000           1200.000000        1200.000000  
mean                  2.194167              4.105000           2.948333  
std                   3.221560              3.541576           0.518866  
min                   0.000000              0.000000           2.000

- from the above we can see that no columns in the dataset are having constant in nature

### showing the statical feature of categorical columns

In [12]:
data.describe(include='O')

EmpNumber Gender EducationBackground MaritalStatus EmpDepartment  \
count       1200   1200                1200          1200          1200   
unique      1200      2                   6             3             6   
top     E1001000   Male       Life Sciences       Married         Sales   
freq           1    725                 492           548           373   

             EmpJobRole BusinessTravelFrequency OverTime Attrition  
count              1200                    1200     1200      1200  
unique               19                       3        2         2  
top     Sales Executive           Travel_Rarely       No        No  
freq                270                     846      847      1022

In [13]:
# showing the total missing values in the dataset
data.isnull().sum()

EmpNumber                       0
Age                             0
Gender                          0
EducationBackground             0
MaritalStatus                   0
EmpDepartment                   0
EmpJobRole                      0
BusinessTravelFrequency         0
DistanceFromHome                0
EmpEducationLevel               0
EmpEnvironmentSatisfaction      0
EmpHourlyRate                   0
EmpJobInvolvement               0
EmpJobLevel                     0
EmpJobSatisfaction              0
NumCompaniesWorked              0
OverTime                        0
EmpLastSalaryHikePercent        0
EmpRelationshipSatisfaction     0
TotalWorkExperienceInYears      0
TrainingTimesLastYear           0
EmpWorkLifeBalance              0
ExperienceYearsAtThisCompany    0
ExperienceYearsInCurrentRole    0
YearsSinceLastPromotion         0
YearsWithCurrManager            0
Attrition                       0
PerformanceRating               0
dtype: int64

- from above we can say that no missing values are present in the dataset

In [14]:
data.duplicated().sum()

0

- no duplicated rows present in the dataset

In [15]:
# filtering the categorical and  numerical columns
categorical_columns = []
numerical_columns = []
for columns in data:
    if data[columns].dtypes == 'object':
        categorical_columns.append(columns)
    else:
        numerical_columns.append(columns)

In [16]:
# showing the categorical columns in the dataset
categorical_columns

['EmpNumber',
 'Gender',
 'EducationBackground',
 'MaritalStatus',
 'EmpDepartment',
 'EmpJobRole',
 'BusinessTravelFrequency',
 'OverTime',
 'Attrition']

In [17]:
# total numerical columns in the dataset
numerical_columns

['Age',
 'DistanceFromHome',
 'EmpEducationLevel',
 'EmpEnvironmentSatisfaction',
 'EmpHourlyRate',
 'EmpJobInvolvement',
 'EmpJobLevel',
 'EmpJobSatisfaction',
 'NumCompaniesWorked',
 'EmpLastSalaryHikePercent',
 'EmpRelationshipSatisfaction',
 'TotalWorkExperienceInYears',
 'TrainingTimesLastYear',
 'EmpWorkLifeBalance',
 'ExperienceYearsAtThisCompany',
 'ExperienceYearsInCurrentRole',
 'YearsSinceLastPromotion',
 'YearsWithCurrManager',
 'PerformanceRating']

In [18]:
# filtering the categorical columns which has string values 
non_encoded_columns = [col for col in data if data[col].dtype ==  'O']
non_encoded_columns

['EmpNumber',
 'Gender',
 'EducationBackground',
 'MaritalStatus',
 'EmpDepartment',
 'EmpJobRole',
 'BusinessTravelFrequency',
 'OverTime',
 'Attrition']

In [19]:
# checking the total count of features in target columns
data.PerformanceRating.value_counts()

PerformanceRating
3    874
2    194
4    132
Name: count, dtype: int64

- From Above we can say that the dataset is unbalanced in nature

In [20]:
# Find the minimum age and corresponding department
min_age_row = data.loc[data['Age'].idxmin()]
min_age = min_age_row['Age']
min_age_department = min_age_row['EmpDepartment']

# Find the maximum age and corresponding department
max_age_row = data.loc[data['Age'].idxmax()]
max_age = max_age_row['Age']
max_age_department = max_age_row['EmpDepartment']

# Print the results
print(f"Minimum Age: {min_age} in Department: {min_age_department}")
print(f"Maximum Age: {max_age} in Department: {max_age_department}")

Minimum Age: 18 in Department: Research & Development
Maximum Age: 60 in Department: Sales


In [21]:
# Group by 'Gender' and calculate the mean of 'EmpJobSatisfaction'
gender_satisfaction = data.groupby('Gender')['EmpJobSatisfaction'].mean()

# Find the gender with the highest mean satisfaction
most_satisfied_gender = gender_satisfaction.idxmax()
highest_satisfaction_mean = gender_satisfaction.max()

# Print the result
print(f"The gender most satisfied with the job is {most_satisfied_gender} with a mean satisfaction of {highest_satisfaction_mean:.2f}")

The gender most satisfied with the job is Male with a mean satisfaction of 2.75


In [22]:
# Convert 'Attrition' column to binary (1 for 'Yes', 0 for 'No')
data['AttritionBinary'] = data['Attrition'].map({'Yes': 1, 'No': 0})

# Group by 'MaritalStatus' and calculate the sum, mean, and min of 'AttritionBinary'
attrition_stats = data.groupby('MaritalStatus')['AttritionBinary'].agg(['sum', 'mean', 'min'])

# Find the marital status with the maximum total attrition
max_attrition_marital_status = attrition_stats['sum'].idxmax()
max_attrition_total = attrition_stats['sum'].max()

# Find the marital status with the highest mean attrition
max_mean_attrition_marital_status = attrition_stats['mean'].idxmax()
max_mean_attrition_mean = attrition_stats['mean'].max()

# Find the marital status with the minimum attrition
min_attrition_marital_status = attrition_stats['min'].idxmin()
min_attrition_min = attrition_stats['min'].min()

# Print the results
print(f"The marital status with the maximum total attrition is {max_attrition_marital_status} with a total of {max_attrition_total} employees.")
print(f"The marital status with the highest mean attrition is {max_mean_attrition_marital_status} with a mean attrition of {max_mean_attrition_mean:.2%}")
print(f"The marital status with the minimum attrition is {min_attrition_marital_status} with a minimum attrition of {min_attrition_min}")

# Drop the temporary 'AttritionBinary' column if not needed
data = data.drop('AttritionBinary', axis=1)

The marital status with the maximum total attrition is Single with a total of 94 employees.
The marital status with the highest mean attrition is Single with a mean attrition of 24.48%
The marital status with the minimum attrition is Divorced with a minimum attrition of 0


In [23]:
# Filter the DataFrame for employees working overtime
overtime_data = data[data['OverTime'] == 'Yes']

# Group by 'Gender' and 'MaritalStatus' and calculate the count
overtime_stats = overtime_data.groupby(['Gender', 'MaritalStatus']).size().reset_index(name='Count')

# Print the results
print("Gender and Marital Status combinations working overtime:")
print(overtime_stats)

Gender and Marital Status combinations working overtime:
   Gender MaritalStatus  Count
0  Female      Divorced     36
1  Female       Married     62
2  Female        Single     52
3    Male      Divorced     49
4    Male       Married     96
5    Male        Single     58


In [24]:
# Group by 'Gender' and 'PerformanceRating' and calculate the count
performance_stats = data.groupby(['Gender', 'PerformanceRating']).size().reset_index(name='Count')

# Print the results
print("Performance Ratings by Gender:")
print(performance_stats)

Performance Ratings by Gender:
   Gender  PerformanceRating  Count
0  Female                  2     75
1  Female                  3    349
2  Female                  4     51
3    Male                  2    119
4    Male                  3    525
5    Male                  4     81


In [25]:
# Group by 'NumCompaniesWorked' and calculate the mean performance rating
performance_by_companies = data.groupby('NumCompaniesWorked')['PerformanceRating'].mean().reset_index()

# Print the results
print("Average Performance Rating by Number of Companies Worked:")
print(performance_by_companies)

Average Performance Rating by Number of Companies Worked:
   NumCompaniesWorked  PerformanceRating
0                   0           2.897436
1                   1           2.942263
2                   2           2.910569
3                   3           2.962406
4                   4           3.084112
5                   5           2.962264
6                   6           2.982143
7                   7           2.950000
8                   8           2.950000
9                   9           2.846154


In [26]:
# Group by 'DistanceFromHome' and calculate the mean performance rating
performance_by_distance = data.groupby('DistanceFromHome')['PerformanceRating'].mean().reset_index()

# Print the results
print("Average Performance Rating by Distance from Home:")
print(performance_by_distance)

Average Performance Rating by Distance from Home:
    DistanceFromHome  PerformanceRating
0                  1           2.958824
1                  2           2.956522
2                  3           3.029851
3                  4           2.980392
4                  5           2.925926
5                  6           2.934783
6                  7           3.000000
7                  8           2.942029
8                  9           2.893939
9                 10           3.000000
10                11           3.040000
11                12           2.937500
12                13           2.857143
13                14           2.941176
14                15           2.764706
15                16           3.160000
16                17           3.000000
17                18           2.750000
18                19           2.944444
19                20           2.789474
20                21           2.933333
21                22           2.823529
22                23          

In [27]:
# Grouping and Aggregation
grouped_data = data.groupby('Attrition')['EmpJobSatisfaction'].mean()

# Hypothesis Testing (t-test)
attrition_yes = data[data['Attrition'] == 'Yes']['EmpJobSatisfaction']
attrition_no = data[data['Attrition'] == 'No']['EmpJobSatisfaction']


# Print the results
print(f"Average Job Satisfaction for Attrition: Yes - {grouped_data['Yes']:.2f}")
print(f"Average Job Satisfaction for Attrition: No - {grouped_data['No']:.2f}")

Average Job Satisfaction for Attrition: Yes - 2.52
Average Job Satisfaction for Attrition: No - 2.77


In [28]:
# Grouping and Aggregation
promotion_by_education = data.groupby('EmpEducationLevel')['YearsSinceLastPromotion'].apply(lambda x: (x > 0).mean()).reset_index(name='PromotionRate')

# Hypothesis Testing (Chi-Square Test)
promotion_matrix = pd.crosstab(data['EmpEducationLevel'], data['YearsSinceLastPromotion'] > 0)

# Print the values
print("Promotion Rates by Education Level:")
print(promotion_by_education)

Promotion Rates by Education Level:
   EmpEducationLevel  PromotionRate
0                  1       0.621622
1                  2       0.606695
2                  3       0.590200
3                  4       0.627329
4                  5       0.642857


In [29]:
# Correlation Analysis
correlation_matrix = data[['EmpHourlyRate', 'EmpJobSatisfaction', 'PerformanceRating']].corr()


# Grouping and Aggregation
rate_vs_satisfaction = data.groupby('EmpHourlyRate')['EmpJobSatisfaction'].agg(['mean', 'max']).reset_index()


# Print the results
print("Correlation Matrix:")
correlation_matrix


Correlation Matrix:


EmpHourlyRate  EmpJobSatisfaction  PerformanceRating
EmpHourlyRate            1.000000           -0.066417          -0.043116
EmpJobSatisfaction      -0.066417            1.000000           0.000606
PerformanceRating       -0.043116            0.000606           1.000000

In [30]:
print("\nAverage and Maximum Job Satisfaction by Hourly Rate:")
rate_vs_satisfaction


Average and Maximum Job Satisfaction by Hourly Rate:


EmpHourlyRate      mean  max
0              30  2.615385    4
1              31  2.615385    4
2              32  3.157895    4
3              33  2.875000    4
4              34  2.666667    4
..            ...       ...  ...
66             96  2.636364    4
67             97  2.888889    4
68             98  2.700000    4
69             99  2.578947    4
70            100  2.642857    4

[71 rows x 3 columns]

In [31]:
# Calculating the mean hourly rate for each department
data.groupby('EmpDepartment')['EmpHourlyRate'].mean()

EmpDepartment
Data Science              62.250000
Development               65.828255
Finance                   66.693878
Human Resources           65.703704
Research & Development    67.023324
Sales                     65.319035
Name: EmpHourlyRate, dtype: float64

Research & Development department has the highest average hourly rate of 67.02, followed by the Finance department with 66.69. The Data Science department has the lowest average hourly rate of 62.25, which is about 5 less than the Research & Development department.

In [32]:
# Count the number of employee in each department on which job role
data.groupby(['EmpDepartment', 'EmpJobRole']).size().reset_index(name='employee_count')

EmpDepartment                 EmpJobRole  employee_count
0             Data Science             Data Scientist              20
1              Development           Business Analyst              16
2              Development           Delivery Manager              12
3              Development                  Developer             236
4              Development           Senior Developer              52
5              Development        Technical Architect               7
6              Development             Technical Lead              38
7                  Finance            Finance Manager              49
8          Human Resources            Human Resources              45
9          Human Resources                    Manager               9
10  Research & Development  Healthcare Representative              33
11  Research & Development      Laboratory Technician              64
12  Research & Development                    Manager               8
13  Research & Development                Manager R&D              94
14  Research & Development     Manufacturing Director              33
15  Research & Development          Research Director              19
16  Research & Development         Research Scientist              77
17  Research & Development         Senior Manager R&D              15
18                   Sales                    Manager              34
19                   Sales            Sales Executive             270
20                   Sales       Sales Representative              69

- The data shows that there are **1200 employees** in total in the company, with **6 different departments** and **21 different job roles**.
- The **Development** department has the most number of employees (**361**), followed by the **Sales** department (**373**). The **Data Science** department has the least number of employees (**20**).
- The **Sales Executive** job role has the most number of employees (**270**), followed by the **Developer** job role (**236**). The **Technical Architect** job role has the least number of employees (**7**).
- The **Development** department has the most diversity of job roles (**6**), ranging from **Business Analyst** to **Technical Lead**. The **Data Science** department has the least diversity of job roles (**1**), which is **Data Scientist**.
- The **Research & Development** department has the most number of managers (**136**), followed by the **Sales** department (**34**). The **Human Resources** department has the least number of managers (**9**).

In [33]:
# Group by 'EmpDepartment' and calculate the average job satisfaction for each department
average_job_satisfaction = data.groupby('EmpDepartment')['EmpJobSatisfaction'].mean().reset_index()
average_job_satisfaction

EmpDepartment  EmpJobSatisfaction
0            Data Science            2.700000
1             Development            2.747922
2                 Finance            2.653061
3         Human Resources            2.555556
4  Research & Development            2.720117
5                   Sales            2.766756

- The **Sales** department has the highest average employee job satisfaction of **2.77**, followed by the **Development** department with **2.75**.
- The **Human Resources** department has the lowest average employee job satisfaction of **2.56**, which is about 0.2 points lower than the Sales department.
- The **Data Science** and **Research & Development** departments have similar average employee job satisfaction of around **2.7**.
- The data shows that there is not much variation in the average employee job satisfaction across different departments, as all of them are close to the middle value of **2.5**. This means that the employees are neither very satisfied nor very dissatisfied with their jobs.

In [34]:
# Read the Excel file into a DataFrame
data = pd.read_excel('https://github.com/PawarMukesh/Employee-Performance-Analysis/raw/mukesh/Employee%20Performance%20Analysis/INX_Future_Inc_Employee_Performance_CDS_Project2_Data_V1.8.xls')

# Group by 'EmpDepartment' and calculate the average performance rating for each department
average_performance_rating = data.groupby('EmpDepartment')['PerformanceRating'].mean().reset_index()
average_performance_rating

EmpDepartment  PerformanceRating
0            Data Science           3.050000
1             Development           3.085873
2                 Finance           2.775510
3         Human Resources           2.925926
4  Research & Development           2.921283
5                   Sales           2.860590

- The **Development** department has the highest average performance rating of **3.09**, followed by the **Data Science** department with **3.05**.
- The **Finance** department has the lowest average performance rating of **2.78**, which is about 0.3 points lower than the Development department.
- The **Human Resources** and **Research & Development** departments have similar average performance ratings of around **2.92**.
- The data shows that there is not much variation in the average performance rating across different departments, as all of them are close to the middle value of **3**. This means that the employees are performing at a similar level in their respective departments.

In [35]:
last_salary_hike = data.groupby('EmpDepartment')['EmpLastSalaryHikePercent'].mean().reset_index()
last_salary_hike

EmpDepartment  EmpLastSalaryHikePercent
0            Data Science                 16.450000
1             Development                 15.204986
2                 Finance                 15.367347
3         Human Resources                 14.740741
4  Research & Development                 15.300292
5                   Sales                 15.152815

- The **Data Science** department has the highest average employee last salary hike percent of **16.45%**, followed by the **Finance** department with **15.37%**.
- The **Human Resources** department has the lowest average employee last salary hike percent of **14.74%**, which is about 1.7 percentage points lower than the Data Science department.
- The **Development** and **Sales** departments have similar average employee last salary hike percent of around **15.2%**.

In [36]:
work_life_balance_count = data.groupby(['EmpDepartment', 'EmpWorkLifeBalance']).size().reset_index(name='Count')
work_life_balance_count

EmpDepartment  EmpWorkLifeBalance  Count
0             Data Science                   1      3
1             Data Science                   2      5
2             Data Science                   3     11
3             Data Science                   4      1
4              Development                   1     25
5              Development                   2     88
6              Development                   3    223
7              Development                   4     25
8                  Finance                   1      2
9                  Finance                   2     14
10                 Finance                   3     31
11                 Finance                   4      2
12         Human Resources                   1      3
13         Human Resources                   2      6
14         Human Resources                   3     37
15         Human Resources                   4      8
16  Research & Development                   1     19
17  Research & Development                   2     90
18  Research & Development                   3    194
19  Research & Development                   4     40
20                   Sales                   1     12
21                   Sales                   2     91
22                   Sales                   3    231
23                   Sales                   4     39

In [37]:
# Filter the DataFrame to include only rows where 'EmpWorkLifeBalance' is 3 or 4
filtered_work_life_balance = work_life_balance_count[work_life_balance_count['EmpWorkLifeBalance'].isin([3, 4])]
filtered_work_life_balance

EmpDepartment  EmpWorkLifeBalance  Count
2             Data Science                   3     11
3             Data Science                   4      1
6              Development                   3    223
7              Development                   4     25
10                 Finance                   3     31
11                 Finance                   4      2
14         Human Resources                   3     37
15         Human Resources                   4      8
18  Research & Development                   3    194
19  Research & Development                   4     40
22                   Sales                   3    231
23                   Sales                   4     39